packages/protocol/test/common/multisig.ts

```diff
@@ -0,0 +1,382 @@
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      txId = txEvent.args.transactionId
+    })
+
+    it('should allow an owner to confirm a transaction', async () => {
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+      assert.isTrue(await multiSig.confirmations(txId, accounts[1]))
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const parsedTxData = parseMultiSigTransaction(await multiSig.transactions(txId))
+      assert.isTrue(parsedTxData.executed)
+    })
+
+    it('should not allow an owner to confirm a transaction twice', async () => {
+      await assertRevert(multiSig.confirmTransaction(txId, { from: accounts[0] }))
+    })
+
+    it('should not allow a non-owner to confirm a transaction', async () => {
+      await assertRevert(multiSig.confirmTransaction(txId, { from: accounts[2] }))
+    })
+  })
+
+  describe('#revokeConfirmation()', () => {
+    let txId: number
+    let tx: string
+    beforeEach(async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.addOwner(accounts[2]).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      txId = txEvent.args.transactionId
+    })
+
+    it('should allow an owner to revoke a confirmation', async () => {
+      await multiSig.revokeConfirmation(txId)
+      assert.isFalse(await multiSig.confirmations(txId, accounts[0]))
+    })
+
+    it('should not allow a non-owner to revoke a confirmation', async () => {
+      await assertRevert(multiSig.revokeConfirmation(txId, { from: accounts[2] }))
+    })
+
+    it('should not allow an owner to revoke before confirming', async () => {
+      await assertRevert(multiSig.revokeConfirmation(txId, { from: accounts[1] }))
+    })
+  })
+
+  describe('#addOwner()', () => {
+    it('should allow a new owner to be added via the MultiSig', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.addOwner(accounts[2]).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      const txId = txEvent.args.transactionId
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+      assert.isTrue(await multiSig.isOwner(accounts[2]))
+    })
+
+    it('should not allow an external account to add an owner', async () => {
```

Comment:
Should this test instead be calling `addOwner` directly rather than `submitTransaction`?

Reply:
changed